yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -540,6 +560,27 @@ private[spark] class Client(
     amContainer
   }
 
+  def setupCredentials(): Unit = {
+    if (args.principal != null) {
+      Preconditions.checkNotNull(
```

Comment:
sorry for flip-flopping here. The methods in `Preconditions` work weirdly in Scala as you've noticed. Should probably use Scala's `require()` here (which throws IllegalArgumentException, which is also a little more correct).